### PR TITLE
Offer opening MapViews in Browser or native Maps app

### PR DESCRIPTION
**Pull request by mueller-ma**
Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.kt [hunk 1]
```diff
@@ -260,3 +261,7 @@ fun MapView.setMarker(
 fun Location.toGeoPoint(): GeoPoint {
     return GeoPoint(this)
 }
+
+fun getUrlForMaps(location: Location): String {
+    return "https://www.openstreetmap.org/#map=16/${location.latitude}/${location.longitude}"
+}
```
Comment (maniac103) [inline mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.kt]: Suggestion: `fun Location?.toMapsUrl(): String?`

File: mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.kt [hunk 2]
```diff
@@ -193,3 +193,7 @@ fun GoogleMap.setMarker(position: LatLng, item: Item, label: CharSequence?, canD
 fun Location.toLatLng(): LatLng {
     return LatLng(latitude, longitude)
 }
+
+fun getUrlForMaps(location: Location): String {
+    return "https://www.google.de/maps/@${location.latitude},${location.longitude},16z"
+}
```
Comment (maniac103) [inline mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.kt]: As above

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt [hunk 3]
```diff
@@ -183,6 +184,10 @@ class WidgetListFragment : Fragment(), WidgetAdapter.ItemClickListener {
                     widget.linkedPage?.let { createShortcut(context, it) }
                     return true
                 }
+                CONTEXT_MENU_ID_OPEN_IN_MAPS -> {
+                    getUrlForMaps(widget.item?.state?.asLocation ?: return true).toUri().openInBrowser(context)
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt]: With the suggestion above, this becomes `widget.item?.state?.asLocation?.toMapsUrl()?.toUri()?.openInBrowser(context)`

Still slightly weird with the large number of question marks, but IMHO better than the return in the middle.

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt]: Imagine doing that in Java xD

**Pull request merged**